drivers/clock_control/clock_stm32_ll_common.c

```diff
@@ -361,6 +364,36 @@ static int stm32_clock_control_init(struct device *dev)
 	/* Disable other clocks */
 	LL_RCC_HSI_Disable();
 	LL_RCC_MSI_Disable();
+#elif CONFIG_CLOCK_STM32_PLL_SRC_PLL2
+	int hse_bypass = LL_UTILS_HSEBYPASS_OFF;
+	u32_t PLL2Frequency;
+	LL_UTILS_PLLInitTypeDef s_PLL2InitStruct;
+
+#ifdef CONFIG_CLOCK_STM32_HSE_BYPASS
+	hse_bypass = LL_UTILS_HSEBYPASS_ON;
+#endif /* CONFIG_CLOCK_STM32_HSE_BYPASS */
+
+	/* get PLL2 configuration */
+	config_pll2_init(&s_PLL2InitStruct);
+
+	PLL2Frequency =
+		(CONFIG_CLOCK_STM32_HSE_CLOCK / CONFIG_CLOCK_STM32_PLL2_PREDIV2)
+		* CONFIG_CLOCK_STM32_PLL2_MULTIPLIER;
+
+	/* switch the input to use PLL2 instead of HSE directly */
+	s_PLLInitStruct.Prediv |= RCC_CFGR2_PREDIV1SRC << 4U;
```

Comment:
I think this could be done directly in `config_pll2_init`.

Reply:
I didn't want to contaminate things too much... it should probably technically live in `config_pll_init()`, though shouldn't be set when the PLL is clocked by HSE (or anything other than PLL2).

The `RCC_CFGR2_PREDIV1SRC << 4U` is a bit _magic_... but comes from here:
- [stm32f1xx_ll_rcc.h:426](https://github.com/zephyrproject-rtos/hal_stm32/blob/master/stm32cube/stm32f1xx/drivers/include/stm32f1xx_ll_rcc.h#L426)
- [stm32f1xx_ll_rcc.h:443](https://github.com/zephyrproject-rtos/hal_stm32/blob/master/stm32cube/stm32f1xx/drivers/include/stm32f1xx_ll_rcc.h#L443)

Perhaps worth ORing [`LL_RCC_PLLSOURCE_PLL2`](https://github.com/zephyrproject-rtos/hal_stm32/blob/master/stm32cube/stm32f1xx/drivers/include/stm32f1xx_ll_rcc.h#L422) over `pllint->Prediv` in [`config_pll_init()`](https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/clock_control/clock_stm32f1.c#L33) if `CONFIG_CLOCK_STM32_PLL_SRC_PLL2` is defined?

Reply:
I think it is ok to put it in `config_pll2_init` since dedicated to pll2, this shouldn't have much impact